lib/membrane/aac/parser/helper.ex

```diff
@@ -171,4 +199,94 @@ defmodule Membrane.AAC.Parser.Helper do
         encapsulation: :none,
         samples_per_frame: if(frame_length_flag == 0, do: 1024, else: 960)
       }
+
+  @spec parse_esds(binary()) ::
+          %{profile: AAC.profile(), samples_per_frame: AAC.samples_per_frame()}
+  def parse_esds(esds) do
+    stream_priority = 0
+
+    {esds, <<>>} = unpack_esds_section(esds, 3)
+    <<_elementary_stream_id::16, ^stream_priority, rest::binary>> = esds
+
+    {section_4, esds_section_6} = unpack_esds_section(rest, 4)
+    {<<2>>, <<>>} = unpack_esds_section(esds_section_6, 6)
+
+    # 64 = mpeg4-audio
+    object_type_id = 64
+    # 5 = audio
+    stream_type = 5
+    upstream_flag = 0
+    reserved_flag_set_to_1 = 1
+    buffer_size = 0
+
+    <<^object_type_id, ^stream_type::6, ^upstream_flag::1, ^reserved_flag_set_to_1::1,
+      ^buffer_size::24, _max_bit_rate::32, _avg_bit_rate::32, esds_section_5::binary>> = section_4
+
+    {section_5, <<>>} = unpack_esds_section(esds_section_5, 5)
+
+    depends_on_core_coder = 0
+    extension_flag = 0
+
+    <<aot_id::5, _frequency_id::4, _channel_config_id::4, frame_length_id::1,
+      ^depends_on_core_coder::1, ^extension_flag::1>> = section_5
+
+    %{
+      profile: AAC.aot_id_to_profile(aot_id),
+      samples_per_frame: AAC.frame_length_id_to_samples_per_frame(frame_length_id)
+    }
+  end
+
+  defp unpack_esds_section(section, section_no) do
+    type_tag = <<128, 128, 128>>
+
+    <<^section_no::8-integer, ^type_tag::binary-size(3), payload_size::8-integer, rest::binary>> =
+      section
+
+    <<payload::binary-size(payload_size), rest::binary>> = rest
+    {payload, rest}
+  end
+
+  @spec generate_esds(AAC.t(), Membrane.Element.state()) :: binary()
+  def generate_esds(stream_format, state) do
```

Comment:
let's move it before `parse_esds`, so that we have some consistent order of functions in the file (now it is: `generate_audio_specifig_config`, `parse_audio_specific_config`, `parse_esds`, `generate_esds`).

Reply:
I would also consider separating the ESDS functions into `Helper.ESDS` and audio_specific_config into `Helper.AudioSpecificConfig`)